src/dex/spectrum.ts

```diff
@@ -0,0 +1,209 @@
+    }
+
+    const builder: DefinitionBuilder = await new DefinitionBuilder().loadDefinition(pool);
+    const datum: DefinitionField = await provider.datumValue(utxo.datumHash);
+    const parameters: DatumParameters = builder.pullParameters(datum as DefinitionConstr);
+
+    liquidityPool.identifier = typeof parameters.PoolIdentifier === 'string' ? parameters.PoolIdentifier : '';
+    liquidityPool.poolFeePercent = typeof parameters.LpFee === 'number' ? (1000 - parameters.LpFee) / 10 : 0.3;
+
+    return liquidityPool;
+  }
+
+  public estimatedGive(liquidityPool: LiquidityPool, swapOutToken: Token, swapOutAmount: bigint): bigint {
+    // TODO.
+    return 0n;
+  }
+
+  public estimatedReceive(liquidityPool: LiquidityPool, swapInToken: Token, swapInAmount: bigint): bigint {
+    // TODO.
+    return 2n;
+  }
+
+  public priceImpactPercent(liquidityPool: LiquidityPool, swapInToken: Token, swapInAmount: bigint): number {
+    // TODO.
+    return 0;
+  }
+
+  public async buildSwapOrder(liquidityPool: LiquidityPool, swapParameters: DatumParameters, spendUtxos: UTxO[] = []): Promise<PayToAddress[]> {
+    const batcherFee: SwapFee | undefined = this.swapOrderFees().find((fee: SwapFee) => fee.id === 'batcherFee');
+    const deposit: SwapFee | undefined = this.swapOrderFees().find((fee: SwapFee) => fee.id === 'deposit');
+    const minReceive = swapParameters.MinReceive as bigint;
+
+    if (!batcherFee || !deposit || !minReceive) {
+      return Promise.reject('Parameters for datum are not set.');
+    }
+
+    const batcherFeeForToken = Number(batcherFee.value) / Number(minReceive);
+    const number = bignumber(batcherFeeForToken.toString());
+    const [numerator, denominator] = decimalToFractional(number);
+
+    const lpfee = 1000 - liquidityPool.poolFeePercent * 10;
+
+    swapParameters = {
+      ...swapParameters,
+      [DatumParameterKey.TokenPolicyId]: liquidityPool.lpToken.policyId,
+      [DatumParameterKey.TokenAssetName]: liquidityPool.lpToken.nameHex,
+      [DatumParameterKey.LpFee]: lpfee,
+      [DatumParameterKey.LpFeeNumerator]: numerator,
+      [DatumParameterKey.LpFeeDenominator]: denominator,
+    };
+
+    const datumBuilder: DefinitionBuilder = new DefinitionBuilder();
+    await datumBuilder.loadDefinition(order).then((builder: DefinitionBuilder) => {
+      builder.pushParameters(swapParameters);
+    });
+
+    return [
+      this.buildSwapOrderPayment(swapParameters, {
+        address: this.orderAddress,
+        addressType: AddressType.Contract,
+        assetBalances: [
+          {
+            asset: 'lovelace',
+            quantity: batcherFee?.value + deposit.value,
+          },
+        ],
+        datum: datumBuilder.getCbor(),
+        spendUtxos: spendUtxos,
+      }),
+    ];
+  }
+
+  public buildCancelSwapOrder(txOutputs: UTxO[], returnAddress: string): Promise<PayToAddress[]> {
+    throw new Error('Method not implemented.');
+  }
+
+  public swapOrderFees(): SwapFee[] {
```

Comment:
We could adjust some things to supply the swap in amount so we can get accurate fees